internal/controller/clusterinstance_controller.go

```diff
@@ -356,22 +373,65 @@ func createOrPatch(
 			}
 		}
 
-		if err := c.Create(ctx, &obj); err != nil {
+		if err := c.Create(ctx, &renderedObj); err != nil {
 			return controllerutil.OperationResultNone, err
 		}
 		return controllerutil.OperationResultCreated, nil
 	}
 
 	// Object exists, update it
-	obj.SetResourceVersion(existingObj.GetResourceVersion())
-	obj.SetOwnerReferences(existingObj.GetOwnerReferences())
-	patch := client.MergeFrom(existingObj)
+	patch := client.MergeFrom(liveObj.DeepCopy())
+	updatedLiveObj := liveObj.DeepCopy()
+
+	// If the resource contains a status then remove it from the unstructured
+	// copy to avoid unnecessary patching later.
+	unstructured.RemoveNestedField(updatedLiveObj.Object, "status")
+
+	// Merge metadata.annotations and metadata.labels
+	updatedLiveObj.SetAnnotations(mergeMaps(renderedObj.GetAnnotations(), liveObj.GetAnnotations()))
+	updatedLiveObj.SetLabels(mergeMaps(renderedObj.GetLabels(), liveObj.GetLabels()))
+
+	// Replace updatedLiveObj.spec fields with those in renderedObj.spec
+	spec, ok := renderedObj.Object["spec"].(map[string]interface{})
+	if !ok {
+		return controllerutil.OperationResultNone, fmt.Errorf("missing spec in rendered install template")
+	}
+	for _, key := range maps.Keys(spec) {
+		nestedField := []string{"spec", key}
+		nestedValue, ok, err := unstructured.NestedFieldCopy(renderedObj.Object, nestedField...)
+		if err != nil {
+			return controllerutil.OperationResultNone, err
+		}
+		if !ok {
+			continue
+		}
+		if err := unstructured.SetNestedField(updatedLiveObj.Object, nestedValue, nestedField...); err != nil {
+			return controllerutil.OperationResultNone, err
+		}
+	}
 
-	if err := c.Patch(ctx, &obj, patch); err != nil {
-		return controllerutil.OperationResultNone, err
+	resourceId := ci.GetResourceId(
+		updatedLiveObj.GetName(),
+		updatedLiveObj.GetNamespace(),
+		updatedLiveObj.GetKind(),
+	)
+	// Compute difference between the liveObject and updatedLiveObject without "status" fields
+	liveObjWithoutStatus := liveObj.DeepCopy()
+	unstructured.RemoveNestedField(liveObjWithoutStatus.Object, "status")
+	if !reflect.DeepEqual(liveObjWithoutStatus, updatedLiveObj) {
+		log.Info(fmt.Sprintf("Change detected in resource %s, updating it", resourceId))
+
+		// Only issue a Patch if the before and after resources (minus status) differ
+		if err := c.Patch(ctx, updatedLiveObj, patch); err != nil {
+			log.Info(fmt.Sprintf("Failed to update resource %s", resourceId))
+			return controllerutil.OperationResultNone, err
+		}
+		return controllerutil.OperationResultUpdated, nil
+	} else {
+		log.Info(fmt.Sprintf("No change detected in resource %s, skipping update request", resourceId))
```

Comment:
This should probably be a debug message. I think it might spam a bit.

Also this else is unnecessary.